WiinUPro/Controls/NintyControl.xaml.cs

```diff
@@ -394,6 +394,32 @@ public void LoadProfile(string fileName)
 
             if (loadedProfile != null)
             {
+                if (fileName.Length > 6 & fileName[fileName.Length - 6] == '_')

```

Comment:
Try using the string Split function instead.

1. Split on the underscore character
2. Take the last part of the split array and remove the file extension
3. Use TryParse to check if it results in a number between 1 and 4

Reply:
Like so?
```C#
string[] subStrings = fileName.Split('_');
if(Int32.TryParse(subStrings[subStrings.Length - 1].Remove(subStrings[subStrings.Length - 1].Length - 4), out int n) && n > 0 && n < 5)
          {
              // Rest of code goes here
          }
```

One might still want to check for the length of the filename of the profile, as "1.wup" for example would otherwise, arguably incorrectly, be recognized as one among other queueable profiles.